modules/stochastic_tools/src/covariances/SquaredExponentialCovariance.C

```diff
@@ -75,20 +81,36 @@ SquaredExponentialCovariance::SquaredExponentialFunction(RealEigenMatrix & K,
   }
 }
 
-void
+bool
 SquaredExponentialCovariance::computedKdhyper(RealEigenMatrix & dKdhp,
                                               const RealEigenMatrix & x,
-                                              std::string hyper_param_name,
+                                              const std::string & hyper_param_name,
                                               unsigned int ind) const
 {
-  if (hyper_param_name == "noise_variance")
+  if (name().length() + 1 > hyper_param_name.length())
+    return false;
+
+  const std::string name_without_prefix = hyper_param_name.substr(name().length() + 1);
+
+  if (name_without_prefix == "noise_variance")
```

Comment:
Is this function used a lot?

String comparisons here seems pretty awful... shouldn't `hyper_param_name` be an enum?

Reply:
The issue is that every covariance function might use a different hyperparameter with a different name.
I did not profile this but I assume the covariance function evaluation (within the conditional) on thousands of samples will be much more expensive than the string comparison here.

Reply:
Sounds good. We could also setup a registry for these if needed in the future. But if it's not a performance concern, I'm fine with it